Screens/Signup.js

```diff
@@ -1,75 +1,132 @@
 import * as firebase from "firebase";
+import "firebase/firestore";
 import { AsyncStorage } from "react-native";
+import { CheckBox } from "react-native-elements";
+import { Checkbox, Switch } from "react-native-paper";
 
 export default class SignUp extends React.Component {
-	state = { email: "", password: "", errorMessage: null };
-	componentDidMount() {
-		//1.
-		// const value=AsyncStorage.gettItem("key");
-		//if (value !== null){}
-		//2.
-		// firebase.auth().onAuthStateChanged(user => {
-		//   if (user != null) {
-		//     console.log("We are authenticated now!");
-		//   }
-		// });
-	}
+    state = {
+        email: "",
+        password: "",
+        isBusiness: false,
+        businessName: "",
+        errorMessage: null,
+    };
 
-	handleSignUp = () => {
-		firebase
-			.auth()
-			.createUserWithEmailAndPassword(this.state.email, this.state.password)
-			.then(() => this.props.navigation.navigate("Main"))
-			.catch(error => this.setState({ errorMessage: error.message }));
-	};
-	render() {
-		return (
-			<View style={styles.container}>
-				<Text>Sign Up</Text>
-				{this.state.errorMessage && (
-					<Text style={{ color: "red" }}>{this.state.errorMessage}</Text>
-				)}
-				<TextInput
-					placeholder="Email"
-					autoCapitalize="none"
-					style={styles.textInput}
-					onChangeText={email => this.setState({ email })}
-					value={this.state.email}
-				/>
-				<TextInput
-					secureTextEntry
-					placeholder="Password"
-					autoCapitalize="none"
-					style={styles.textInput}
-					onChangeText={password => this.setState({ password })}
-					value={this.state.password}
-				/>
-				<Button title="Sign Up" onPress={this.handleSignUp} />
-				<Button
-					title="Already have an account? Login"
-					onPress={() => this.props.navigation.navigate("Login")}
-				/>
-				<Button
-					title="Login via OTP"
-					onPress={() => this.props.navigation.navigate("Otp")}
-				/>
-			</View>
-		);
-	}
+    componentDidMount() {
+        //1.
+        // const value=AsyncStorage.gettItem("key");
+        //if (value !== null){}
+        //2.
+        // firebase.auth().onAuthStateChanged(user => {
+        //   if (user != null) {
+        //     console.log("We are authenticated now!");
+        //   }
+        // });
+    }
+
+    handleSignUp = () => {
+        // Check if business switch is toggled and no business name is given.
+        // Return error if that is the case
+        if (this.state.isBusiness == true && this.state.businessName === "") {
+            this.setState({ errorMessage: "Business name missing" });
+        } else {
+            // create a user with email and password input
+            firebase
+                .auth()
+                .createUserWithEmailAndPassword(
+                    this.state.email,
+                    this.state.password
+                )
+                .then(() => this.props.navigation.navigate("Login"))
```

Comment:
After signing up, do you want navigate back to Login screen or Home screen?

Reply:
Navigating to Profile